src/MultiMessageSender.sol

```diff
@@ -78,69 +86,81 @@ contract MultiMessageSender {
     /// @param _target is the contract address on the destination chain.
     /// @param _callData is the data to be sent to _target by low-level call(eg. address(_target).call(_callData)).
     /// @param _expiration is the unix time when the message expires, zero means never expire.
-    function remoteCall(
-        uint64 _dstChainId,
-        address _multiMessageReceiver,
-        address _target,
-        bytes calldata _callData,
-        uint64 _expiration
-    ) external payable onlyCaller {
-        MessageStruct.Message memory message =
-            MessageStruct.Message(_dstChainId, nonce, _target, _callData, _expiration, "");
-        bytes memory data;
-        uint256 totalFee;
+    function remoteCall(uint256 _dstChainId, address _target, bytes calldata _callData, uint64 _expiration)
+        external
+        payable
+        onlyCaller
+    {
+        /// @dev writes to memory for gas saving
+        address[] memory adapters = senderAdapters[_dstChainId];
+        uint256 adapterLength = adapters.length;
+
+        if (adapterLength == 0) {
+            revert Error.NO_SENDER_ADAPTER_FOUND();
+        }
 
-        uint256 adaptersChainId = 0;
+        /// @dev increments nonce
+        ++nonce;
 
-        /// default adapters
-        if (senderAdapters[_dstChainId].length > 0) {
-            /// if different set of adapters are configured for this desitnation chain
-            adaptersChainId = _dstChainId;
-        }
-        address[] storage adapters = senderAdapters[adaptersChainId];
-        /// send copies of the message through multiple bridges
-        for (uint256 i; i < adapters.length; ++i) {
-            message.bridgeName = IBridgeSenderAdapter(adapters[i]).name();
-            data = abi.encodeWithSelector(IMultiMessageReceiver.receiveMessage.selector, message);
-            uint256 fee =
-                IBridgeSenderAdapter(adapters[i]).getMessageFee(uint256(_dstChainId), _multiMessageReceiver, data);
-            /// if one bridge is paused it shouldn't halt the process
+        MessageLibrary.Message memory message =
+            MessageLibrary.Message(_dstChainId, _target, nonce, _callData, _expiration, "");
+
+        for (uint256 i; i < adapterLength;) {
+            IBridgeSenderAdapter bridgeAdapter = IBridgeSenderAdapter(adapters[i]);
+            message.bridgeName = bridgeAdapter.name();
+
+            /// @dev assumes CREATE2 deployment for mma sender & receiver
+            uint256 fee = bridgeAdapter.getMessageFee(_dstChainId, gac.getMultiMessageReceiver(), abi.encode(message));
+
+            /// @dev if one bridge is paused, the flow shouldn't be broken
             try IBridgeSenderAdapter(adapters[i]).dispatchMessage{value: fee}(
-                uint256(_dstChainId), _multiMessageReceiver, data
-            ) {
-                totalFee += fee;
-            } catch {
+                _dstChainId, gac.getMultiMessageReceiver(), abi.encode(message)
```

Comment:
wouldn't the `gac.getMultiMessageReceiver()` require a destination chain ID parameter here, given this call is being done on the sender chain (which needs to know which destination receiver address to use) and not (by an adapter) on the receiver chain?